src/Tutor.Core/LearnerModel/DomainOverlay/KcMasteryService.cs

```diff
@@ -49,23 +51,31 @@ public Result<List<InstructionalItem>> GetInstructionalItems(int knowledgeCompon
             var kcMastery = _kcMasteryRepository.GetFullKcMastery(knowledgeComponentId, learnerId);
             if (kcMastery == null) return Result.Fail("Learner not enrolled in KC: " + knowledgeComponentId);
 
-            var result = kcMastery.SelectInstructionalItems();
+            var result = kcMastery.RecordInstructionalItemSelection();
+            if (result.IsFailed) return result.ToResult<List<InstructionalItem>>();
+
             _kcMasteryRepository.UpdateKcMastery(kcMastery);
 
-            return result;
+            return Result.Ok(kcMastery.KnowledgeComponent.InstructionalItems.OrderBy(i => i.Order).ToList());
         }
 
         public Result<AssessmentItem> SelectSuitableAssessmentItem(int knowledgeComponentId, int learnerId)
         {
             var kcMastery = _kcMasteryRepository.GetFullKcMastery(knowledgeComponentId, learnerId);
             if (kcMastery == null) return Result.Fail("Learner not enrolled in KC: " + knowledgeComponentId);
 
-            var result = kcMastery.SelectAssessmentItem(_assessmentItemSelector);
+            var selectionResult = _assessmentItemSelector.SelectSuitableAssessmentItemId(kcMastery.AssessmentItemMasteries, kcMastery.IsPassed);
+            if (selectionResult.IsFailed) return selectionResult.ToResult<AssessmentItem>();
+
+            var masteryResult = kcMastery.RecordAssessmentItemInteraction(new AssessmentItemSelected()
+            {
+                AssessmentItemId = selectionResult.Value
+            });
+            if (masteryResult.IsFailed) return masteryResult.ToResult<AssessmentItem>();
```

Comment:
Is making the service smarter smart?

Reply:
I don't see how this makes the service smarter, it just fails if the call to <code>RecordAssessmentItemInteraction</code> did?